.github/workflows/cygwin.yml

```diff
@@ -0,0 +1,55 @@
+name: Build cygwin
+
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      cygwin_branch:
+        description: 'Cygwin branch'
+        required: true
+        default: 'main'
+  workflow_call:
+    inputs:
+      cygwin_branch:
+        type: string
```

Comment:
what about using the same default branch?

Reply:
Added.